evalml/tests/automl_tests/test_automl_search_regression.py

```diff
@@ -305,3 +309,42 @@ def test_automl_supports_time_series_regression(mock_fit, mock_score, X_y_regres
 
         assert result['parameters']['Delayed Feature Transformer'] == configuration
         assert result['parameters']['pipeline'] == configuration
+
+
+@patch('evalml.pipelines.RegressionPipeline.fit')
+@patch('evalml.pipelines.RegressionPipeline.score')
+def test_automl_pickle_generated_pipeline(mock_regression_score, mock_regression_fit, X_y_regression):
+    class RegressionPipelineCustoms(RegressionPipeline):
+        custom_name = "Custom Regression Name"
+        component_graph = ["Imputer", "Linear Regressor"]
+        custom_hyperparameters = {"Imputer": {"numeric_impute_strategy": "most_frequent"}}
+
+    X, y = X_y_regression
+    pipeline = GeneratedPipelineRegression
+
+    a = AutoMLSearch(X_train=X, y_train=y, problem_type='regression')
+    a.search()
+    a.add_to_rankings(RegressionPipelineCustoms({}))
+    seen_name = False
+    for i, row in a.rankings.iterrows():
+        automl_pipeline = a.get_pipeline(row['id'])
+        assert automl_pipeline.__class__ == pipeline
+        assert pickle.loads(pickle.dumps(automl_pipeline))
+        if automl_pipeline.custom_name == RegressionPipelineCustoms.custom_name:
```

Comment:
Testing here to check that `custom_name` and other attributes appear as expected.